### ISSUE TITLE
Inline CONSTRAINTs are not supported

### ISSUE THREAD

**Issue by jeroenbu** [jonwagner/Insight.Database.Schema#38]
Hi,

I'm trying to get the following script (on SQL Server 2008R2) to execute, but my application fails on 
the Load-method of SchemaObjectCollection with the error "Error parsing Sql: [dbo].[Gebruikers] : Inline CONSTRAINTs are not supported".
I'm not sure what I'm doing wrong in the SQL Query; would appreciate it if I got some advice!
Thanks,

Jeroen

```
create table [dbo].[Gebruikers]
    (
     [Id] [int] identity(1, 1)
                not null
    ,[Naam] [varchar](100) not null
    ,[IsGroep] [bit] not null
    ,[Code] [varchar](100) null
    ,[SpecialistId] [int] null
    )
on  [PRIMARY]
go

alter table [Gebruikers] with check add constraint [PK_Gebruikers] primary key clustered ([Id] asc) with (pad_index = off, statistics_norecompute = off, ignore_dup_key = off, allow_row_locks = on, allow_page_locks = on) on [PRIMARY]
go

alter table [Gebruikers] with check add  constraint [FK_Gebruikers_Specialisten] foreign key([SpecialistId]) 
references [Specialisten] ([Id])
go

alter table [Gebruikers] with check check constraint [FK_Gebruikers_Specialisten]
go

--AUTOPROC all Gebruikers Single=Gebruiker Plural=Gebruikers
--GO
```

**Comment by jonwagner** [comment]
I haven't been able to reproduce your exact problem. The exception should only be thrown if your CREATE TABLE also has an inline constraint (e.g. primary key) and it doesn't.

I have found a few other issues:

* The "All" in "AUTOPROC All" should be capitalized. The enum parser is case-sensitive at this point.
* The library is having an issue with the CHECK CHECK CONSTRAINT. It's trying to treat it as a one-time script (which it should), but there's a bug with the way it names its internal script.

I have a fix for the second one, but I'll have to find some time to thoroughly test it. In the meantime, can you take out that one and see if it solves your issue?

**Comment by jeroenbu** [comment]
Hi Jon,

Thanks for replying so soon!

It seems I have misinformed you a bit; my apologies. I did get the error, but not during normal use of the Load-method of SchemaObjectCollection. I *did* get an error there, just not this one. And while investigating, I got the error in my original post.

Full disclosure:
1. The error "Error parsing Sql: [dbo].[Gebruikers] : Inline CONSTRAINTs are not supported" is thrown in this code, at new SchemaObject(sql), evidently during the SqlParser.Parsers call:
```
SchemaObjectCollection schema = new SchemaObjectCollection();
            Assembly assembly = Assembly.GetExecutingAssembly();
            string[] strings = assembly.GetManifestResourceNames();
            Stream stream = assembly.GetManifestResourceStream(strings[0]); //I know this one is the embedded resource with the sql
            using (StreamReader streamReader = new StreamReader(stream))
            {
                string sql = streamReader.ReadToEnd();
                SchemaObject schemaObject = new SchemaObject(sql);
            }
```
By going through the sourcefiles I tried to track down why the code mentioned next wouldn't run.
2. The error I get here 
```
SchemaObjectCollection schema = new SchemaObjectCollection();
Assembly assembly = Assembly.GetExecutingAssembly();
schema.Load(assembly);
```
is System.ArgumentNullException {"Value cannot be null.\r\nParameter name: input"}
with this callstack:
```
   at System.Text.RegularExpressions.Regex.Matches(String input)
   at Insight.Database.Schema.SqlName.SplitSqlName(String fullName) in c:\projects.net\Insight\Insight.Database.Schema\Insight.Database.Schema\SqlName.cs:line 97
   at Insight.Database.Schema.SqlName..ctor(String fullName, Int32 expectedParts) in c:\projects.net\Insight\Insight.Database.Schema\Insight.Database.Schema\SqlName.cs:line 58
   at Insight.Database.Schema.Implementation.SchemaImpl..ctor(String name, String sql, Int32 expectedPartsInName) in c:\projects.net\Insight\Insight.Database.Schema\Insight.Database.Schema\Implementation\SchemaImpl.cs:line 23
   at Insight.Database.Schema.Implementation.Script..ctor(String name, String sql) in c:\projects.net\Insight\Insight.Database.Schema\Insight.Database.Schema\Implementation\Script.cs:line 14
   at Insight.Database.Schema.Implementation.SchemaImpl.GetImplementation(SchemaObjectType type, String name, String sql) in c:\projects.net\Insight\Insight.Database.Schema\Insight.Database.Schema\Implementation\SchemaImpl.cs:line 124
   at Insight.Database.Schema.SchemaObject..ctor(String sql) in c:\projects.net\Insight\Insight.Database.Schema\Insight.Database.Schema\SchemaObject.cs:line 50
   at Insight.Database.Schema.SchemaObjectCollection.Add(String sql) in c:\projects.net\Insight\Insight.Database.Schema\Insight.Database.Schema\SchemaObjectCollection.cs:line 204
   at Insight.Database.Schema.SchemaObjectCollection.Load(TextReader textReader) in c:\projects.net\Insight\Insight.Database.Schema\Insight.Database.Schema\SchemaObjectCollection.cs:line 129
   at Insight.Database.Schema.SchemaObjectCollection.Load(Assembly assembly, Predicate`1 resourceFilter) in c:\projects.net\Insight\Insight.Database.Schema\Insight.Database.Schema\SchemaObjectCollection.cs:line 185
   at Insight.Database.Schema.SchemaObjectCollection.Load(Assembly assembly) in c:\projects.net\Insight\Insight.Database.Schema\Insight.Database.Schema\SchemaObjectCollection.cs:line 154
```

Anyway, I followed your suggestion. When removing the last scriptline from my script (the CHECK CHECK CONSTRAINT), the load is succesfull.

However, another error pops up, I'll investigate that one, and open a new issue if I can't solve that one.

Thx,

Jeroen

**Comment by jonwagner** [comment]
Then it looks like my change will fix your original issue. I'll find some time to check in the code and then we'll work through your outstanding issues.

**Comment by jonwagner** [comment]
I pushed a code fix for this.

**Comment by jonwagner** [comment]
This is fixed in v2.2.10
